src/get-files.ts

```diff
@@ -15,15 +15,15 @@ import {ILog} from "./types";
  */
 export default async function getFiles(
     includeGlobs: ReadonlyArray<string>,
-    explicitExcludeGlobs: ReadonlyArray<string>,
+    excludeGlobs: ReadonlyArray<string>,
     ignoreFileGlobs: ReadonlyArray<string>,
-    log?: ILog,
+    log: ILog,
 ): Promise<Array<string>> {
-    const ignoreFileExcludeGlobs = await ignoreFileGlobsToExcludeGlobs(
+    // We turn all the ignore files we're using into a single predicate that
+    // returns true if a file is allowed, or false if it is ignored.
+    const allowPredicate = await ignoreFileGlobsToAllowPredicate(
```

Comment:
**note:** Instead of using exclude globs, we now have a predicate that we have to run over files after our glob step.